src/io/pti.jl

```diff
@@ -568,16 +568,28 @@ end
 
 
 """
-    parse_pti(filename)
+    parse_pti(filename::String)
 
 Open PTI raw file given by `filename`, passing the file contents as a string
 to the main PTI parser, returning a `Dict` of all the data parsed into the
 proper types.
 """
 function parse_pti(filename::String)::Dict
-    data_string = readstring(open(filename))
+    open(filename) do f
+        parse_pti(f; filename = filename)
+    end
+end
+
+
+"""
+    parse_pti(io::IOStream)
+
+Parses PTI data in `io::IOStream`, returning a `Dict` of all the data into the
+proper types.
+"""
+function parse_pti(io::IOStream; filename::String = "")::Dict
+    data_string = readstring(io)
     pti_data = parse_pti_data(data_string, get_pti_sections())
     pti_data["CASE IDENTIFICATION"][1]["NAME"] = match(r"[\/\\]*(?:.*[\/\\])*(.*)\.raw", lowercase(filename)).captures[1]
```

Comment:
Make sure to test the steam case here, not clear if this regex will pass with the default value.

Reply:
It definitely won't. `match` returns Void if it can't find any matches. But as I commented elsewhere, that can be replaced with 

`pti_data["CASE IDENTIFICATION"][1]["NAME"] = match(r"[\/\\]*(?:.*[\/\\])*(.*)\.raw", lowercase(io.name)).captures[1]`

Reply:
https://github.com/lanl-ansi/PowerModels.jl/pull/311/commits/72478ae2b17a22583db346c01655f948b86dcf3d#diff-8e131837b7a9f19606ba3595ddaccfc8R593